doc/source/remote_reading.rst

```diff
@@ -0,0 +1,166 @@
+
+    {
+        "s3": {
+            "client_kwargs": {"endpoint_url": "https://PLACE-YOUR-SERVER-URL-HERE"},
+            "secret": "VERYBIGSECRET",
+            "key": "ACCESSKEY"
+        }
+    }
+
+
+Using multiple readers
+======================
+
+If multiple readers are used and the required credentials differ, the storage options are passed per reader like this::
+
+    reader1_filenames = [...]
+    reader2_filenames = [...]
+    filenames = {
+        'reader1': reader1_filenames,
+        'reader2': reader2_filenames,
+    }
+    reader1_storage_options = {...}
+    reader2_storage_options = {...}
+    reader_kwargs = {
+        'reader1': {
+            'option1': 'foo',
+            'storage_options': reader1_storage_options,
+        },
+        'reader2': {
+            'option1': 'foo',
+            'storage_options': reader1_storage_options,
+        }
+    }
+    scn = Scene(filenames=filenames, reader_kwargs=reader_kwargs)
+
+
+Caching the remote files
+========================
+
+Caching the remote file locally can speedup the overall processing time significantly, especially if the data are re-used
+for example when testing. The caching can be done by taking advantage of the `fsspec caching mechanism
+<https://filesystem-spec.readthedocs.io/en/latest/features.html#caching-files-locally>`_::
+
+    reader_kwargs = {
+        'storage_options': {
+            's3': {'anon': True},
+            'filecache': {
+                'cache_storage': '/tmp/s3_cache',
+            }
+        }
+    }
+
+    filenames = ['filecache::s3://noaa-goes16/ABI-L1b-RadC/2019/001/17/*_G16_s20190011702186*']
+    scn = Scene(reader='abi_l1b', filenames=filenames, reader_kwargs=reader_kwargs)
+    scn.load(['true_color_raw'])
+    scn2 = scn.resample(scn.coarsest_area(), resampler='native')
+    scn2.save_datasets(base_dir='/tmp/', tiled=True, blockxsize=512, blockysize=512, driver='COG', overviews=[])
+
+
+The following table shows the timings for running the above code with different cache statuses::
+
+.. _cache_timing_table:
+
+.. list-table:: Processing times without and with caching
+    :header-rows: 1
+    :widths: 40 30 30
+
+    * - Caching
+      - Elapsed time
+      - Notes
+    * - No caching
+      - 650 s
+      - remove `reader_kwargs` and `filecache::` from the code
+    * - File cache
+      - 66 s
+      - Initial run
+    * - File cache
+      - 13 s
+      - After the inital run
+
+.. note::
+
+    The cache is not cleaned by Satpy nor fsspec so the user should handle cleaning excess files from `cache_storage`.
+
+
+Resources
+=========
+
+See :class:`~satpy.readers.FSFile` for direct usage of `fsspec` with Satpy, and
+`fsspec documentation <https://filesystem-spec.readthedocs.io/en/latest/index.html>`_ for more details on connection options
+and detailes.
+
+
+Supported readers
+=================
+
+.. _reader_table:
+
+.. list-table:: Satpy Readers capable of reading remote files using `fsspec`
```

Comment:
In a future PR I'll really need to include this information in a big table of all the readers.

Reply:
Forgot: @BENR0 started on that front in https://github.com/pytroll/satpy/pull/1547 but the PR has been at draft stage since Feb 2021.